### PR TITLE
Delete recipe endpoint

### PR DESCRIPTION
**Pull request by Ionel14**

### CODE REVIEW

File: recipe-box-backend/src/main/java/com/endava/recipebox/service/impl/RecipeServiceImpl.java [hunk 1]
```diff
@@ -172,4 +172,22 @@ public String updateRecipe(RecipeEditRequestDTO recipeAddRequestDTO, Long userId
 
         return "Recipe update successfully!";
     }
+
+    @Override
+    public String deleteRecipe(Long recipeId, Long userId) {
+        Recipe recipe = getRecipeById(recipeId);
```
Comment (en-alexandra-costache) [inline recipe-box-backend/src/main/java/com/endava/recipebox/service/impl/RecipeServiceImpl.java:178]: What happens if the recipe is not found?

Comment (Ionel14) [reply recipe-box-backend/src/main/java/com/endava/recipebox/service/impl/RecipeServiceImpl.java:178]: getRecipeById is this method where I assure to throw an error if the recipe is not found.
`    public Recipe getRecipeById(Long recipeId) {
        Optional<Recipe> recipeOptional = recipeRepository.findById(recipeId);
        if (recipeOptional.isEmpty()) {
            throw new BadRequestException("Recipe with ID " + recipeId + " not found.");
        }
        return recipeOptional.get();
    }`

File: recipe-box-backend/src/main/java/com/endava/recipebox/service/impl/RecipeServiceImpl.java [hunk 2]
```diff
@@ -142,10 +142,10 @@ public RecipeDetailsDTO getDetailedRecipeById(Long recipeId, Long userId) {
     @Transactional
     public String updateRecipe(RecipeEditRequestDTO recipeAddRequestDTO, Long userId){
         Recipe recipe = recipeRepository.findById(recipeAddRequestDTO.getId())
-                .orElseThrow(() -> new EntityNotFoundException("Recipe not found"));
+                .orElseThrow(() -> new BadRequestException("Recipe not found."));
 
         User user = userRepository.findById(userId)
-                .orElseThrow(() -> new EntityNotFoundException("User not found"));
+                .orElseThrow(() -> new BadRequestException("User not found."));
```
Comment (en-madalina-marales) [inline recipe-box-backend/src/main/java/com/endava/recipebox/service/impl/RecipeServiceImpl.java:148]: Why did you change the exception type?

Comment (en-madalina-marales) [reply recipe-box-backend/src/main/java/com/endava/recipebox/service/impl/RecipeServiceImpl.java:148]: Bad request is mapped to 400, any http status with 4xx - represents an error produced by caller

**Review by en-madalina-marales — CHANGES_REQUESTED** [review]

File: recipe-box-backend/src/main/java/com/endava/recipebox/service/impl/RecipeServiceImpl.java (hunk 2 above)
Comment (Ionel14) [reply recipe-box-backend/src/main/java/com/endava/recipebox/service/impl/RecipeServiceImpl.java:148]: I changed the exception because that EntityNotFoundException is not a custom exception and I get Internal Server error 500 in postman, with my exception I am informed corectly about the error

**Review by en-madalina-marales — APPROVED** [review]
